AdminWebsite/AdminWebsite/ClientApp/src/app/booking/services/email-pattern-resolver.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { from, Observable } from 'rxjs';
+import { ConfigService } from 'src/app/services/config.service';
+
+@Injectable()
+export class EmailPatternResolver implements Resolve<string> {
+    constructor(private configService: ConfigService) {}
+
+    resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<string> {
+        return from(this.getData());
+    }
+
+    async getData() {
+        const settings = await this.configService.getClientSettings().toPromise();
```

Comment:
will this work?

Reply:
the promise may not be resolved?

Reply:
@phillipjchambers

Reply:
It does, I tested it

Reply:
The route will not be changed _until_ this route is resolved